Fix Adaptive Tone Map issue on very dark (initial) scene

mgreter opened the pull request:
When last luminance is zero and time delta also very small,
chances are that the output luminance is still zero on the
next run, even when current luminance is rather high. The
reason behind is that luminance value is stored on the red
channel of a 1x1 texture. Therefore computed values below
~0.002 get rounded down to zero again (1/255 => ~0.004).

There is also no reason to write to all color channels when
only the red channel is used (ToneMapShader uses red too).

Additionally implements `renderToScreen` composer flag.

mgreter commented:
After reviewing the issue again I found that there is a deeper "problem" in the [ToneMapShader][1]:

```glsl
float fLumAvg = texture2D(luminanceMap, vec2(0.5, 0.5)).r;
...
float fLumScaled = (fLumPixel * middleGrey) / fLumAvg;
```

As you can see there is a chance of a division by zero (if the red channel is zero). I could add another EPSILON there or any other reasonable minimal value, but I believe that adding a `minLuminance` (there already is a `maxLuminance`) might be the best solution? But not sure if other caluclations should also take note of this new option (like they do for max). Will update this PR "soonish" to also address that issue. But I thought somebody might have some advices here?

[1]: https://github.com/mrdoob/three.js/blob/dev/examples/js/shaders/ToneMapShader.js#L59

mrdoob commented:
/ping @MiiBond

MiiBond commented:
A `minLuminance` sounds like a good solution to me. It's been a long time since I looked at this code. I remember that this was pretty messy though. It could probably use a thorough rewrite :)

WestLangley commented:
FYI, this appears to be the updated link referenced in the code:

Photographic Tone Reproduction for Digital Images by Erik Reinhard, et al.
http://www.cs.utah.edu/~reinhard/cdrom/tonemap.pdf
http://www.cis.rit.edu/people/faculty/ferwerda/publications/sig02_paper.pdf

mgreter committed:
Fix Adaptive Tone Map issue on very dark (initial) scene

When last luminance is zero and time delta also very small,
chances are that the output luminance is still zero on the
next run, even when current luminance is rather high. The
reason behind is that luminance value is stored on the red
channel of a 1x1 texture. Therefore computed values below
~0.002 get rounded down to zero again (1/255 => ~0.004).

This could lead to a division by zero in the ToneMapShader.
Added `minLuminance` option to avoid this edge case.

There is also no reason to write to all color channels when
only the red channel is used (ToneMapShader uses red too).

Additionally implements `renderToScreen` composer flag.

mgreter commented:
Thanks for the feedback. I've added a simple implementation for `minLuminance` option. This would be enough to fix the issue I was having. Or do you guys think it is missing anything?

mrdoob commented:
Thanks!
